src/cuda/device_conj.cu

```diff
@@ -0,0 +1,104 @@
+#include "blas/device.hh"
+#include "thrust/complex.h"
```

Comment:
This isn't using thrust, is it? We can remove the include here and the `using` below. To avoid rebasing all the branches, let's do in another PR.